Tests/FoundationEssentialsTests/AttributedString/AttributedStringConstrainingBehaviorTests.swift

```diff
@@ -257,7 +261,7 @@ class TestAttributedStringConstrainingBehavior: XCTestCase {
         
         result = str
         result.characters.replaceSubrange(result.index(result.startIndex, offsetByCharacters: 8) ..< result.index(result.startIndex, offsetByCharacters: 15), with: "Test\nReplacement")
-        verify(string: result, matches: [("Hello, wTest\n", 1), ("Replacementxt Paragraph", 1)], for: \.testParagraphConstrained)
+        verify(string: result, matches: [("Hello, wTest\n", 1), ("Replacementxt Paragraph", 2)], for: \.testParagraphConstrained)
```

Comment:
This is an intentional change in behavior that I feel was previously incorrect. When replacing text across a paragraph boundary, we shouldn't overwrite the second paragraph with the first paragraph's attribute value, instead the two paragraphs should maintain their individual attribute values. This is unlike replacing the subrange with an `AttributedString` where the client supplies attributes that should extend from the replacement into the rest of the ending paragraph